resolver/type_syntax.cc

```diff
@@ -268,10 +261,15 @@ ParsedSig parseSigWithSelfTypeParams(core::MutableContext ctx, ast::Send *sigSen
                         break;
                     }
 
-                    int i = 0;
-                    for (auto &key : hash->keys) {
-                        auto &value = hash->values[i++];
-                        auto lit = ast::cast_tree<ast::Literal>(key);
+                    if (send->hasKwSplat()) {
+                        // TODO(trevor) add an error for this
+                    }
```

Comment:
I was thinking about leaving this for a follow-up PR, but would be happy to finish it now as well.

Reply:
Fine either way.